runtime/gc_base/IndexableObjectAllocationModel.cpp

```diff
@@ -343,49 +347,58 @@ MM_IndexableObjectAllocationModel::getSparseAddressAndDecommitLeaves(MM_Environm
 	const uintptr_t regionSize = extensions->heapRegionManager->getRegionSize();
 
 	uintptr_t byteAmount = 0;
-	UDATA arrayReservedRegionCount = MM_Math::roundToCeiling(regionSize, _dataSize) / regionSize;
 
 	/* Determine how many bytes to allocate outside of the spine (in arraylet leaves). */
 	Assert_MM_true(_allocateDescription.getBytesRequested() >= _allocateDescription.getContiguousBytes());
 	uintptr_t bytesRemaining = _allocateDescription.getBytesRequested() - _allocateDescription.getContiguousBytes();
 
-	/* Allocate leaf for each arraylet and attach it to its leaf pointer in the spine. */
-	uintptr_t arrayoidIndex = 0;
+	uintptr_t arrayReservedRegionCount = 0;
+	uintptr_t fraction = 0;
+	MM_EnvironmentVLHGC *env = MM_EnvironmentVLHGC::getEnvironment(envBase);
+	MM_AllocationContextBalanced *commonContext = (MM_AllocationContextBalanced *)env->getCommonAllocationContext();
+
 	Trc_MM_getSparseAddressAndDecommitLeaves_Entry(envBase->getLanguageVMThread(), spine, (void *)bytesRemaining, arrayReservedRegionCount, (void *)regionSize);
 	while (0 < bytesRemaining) {
 		/* Allocate the next arraylet leaf - leaves are allocated solely for the purpose of
 		   decommitting the memory later on in this function. */
-		void *leaf = envBase->_objectAllocationInterface->allocateArrayletLeaf(
-				envBase, &_allocateDescription, _allocateDescription.getMemorySpace(), true);
-
-		/* If leaf allocation failed set the result to NULL and return. */
-		if (NULL == leaf) {
-			Trc_MM_allocateAndConnectNonContiguousArraylet_leafFailure(envBase->getLanguageVMThread());
-			_allocateDescription.setSpine(NULL);
-			spine = NULL;
-			break;
+		void *leaf = NULL;
+		bool needAllocateLeaf = true;
+		if (regionSize > bytesRemaining) {
+			fraction = bytesRemaining;
+			if (!commonContext->needAllocateReservedRegionFraction(envBase, fraction)) {
```

Comment:
put a short comment like: For code simplicity and lower fragmentation, we always use Common Context.